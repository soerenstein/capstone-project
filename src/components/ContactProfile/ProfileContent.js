import React, { useState } from 'react'
import styled from 'styled-components/macro'

export default function ProfileContent({contacts}) {
  const [isCategoryVisible, setIsCategoryVisible] = useState(false)
  const [isNoteVisible, setIsNoteVisible] = useState(false)
  const {firstName, lastName, company, phone, mail, street, zip, city, category, note} = contacts
  
  return (
    <ProfileWrapper>
      <NameStyled>{firstName + ' ' + lastName}</NameStyled>
      <CompanyStyled>{company}</CompanyStyled>
      <DetailListStyled>
        <DetailItemHeading>Kontaktdaten</DetailItemHeading>
        <DetailItemContainer>
          <DetailItemContent>{phone}</DetailItemContent>
          <DetailItemContent>{mail}</DetailItemContent>
        </DetailItemContainer>
        <DetailItemHeading>Adresse</DetailItemHeading>
        <DetailItemContainer>
          <DetailItemContent>{street}</DetailItemContent>
          <DetailItemContent>{zip + ' ' + city}</DetailItemContent>
        </DetailItemContainer>
        <DetailItemHeading onClick={toggleCategory}>
          Kategorie
        </DetailItemHeading>
        {isCategoryVisible ? (
          <DetailItemContainer>
            <DetailItemContent>{category}</DetailItemContent>
          </DetailItemContainer>
        ) : null}

        <DetailItemHeading onClick={toggleNote}>Notiz</DetailItemHeading>
        {isNoteVisible ? (
          <DetailItemContainer>
            <DetailItemContent>{note}</DetailItemContent>
          </DetailItemContainer>
        ) : null}
      </DetailListStyled>
    </ProfileWrapper>
  )

  function toggleCategory() {
    setIsCategoryVisible(!isCategoryVisible)
  }
  function toggleNote() {
    setIsNoteVisible(!isNoteVisible)
  }
}

const ProfileWrapper = styled.div`
  z-index: 100;
  position: fixed;
  bottom: 0;
  left: 5px;
  right: 5px;
  margin: 0 auto;
  padding: 10px;
  background: rgb(255, 171, 29);
  background: linear-gradient(
    240deg,
    rgba(255, 171, 29, 1) 0%,
    rgba(248, 80, 28, 1) 100%
  );
  height: 90vh;
  z-index: 100;
  border-top-left-radius: 40px;
  border-top-right-radius: 40px;
`

const NameStyled = styled.h2`
  color: white;

  text-align: center;
  margin: 25px 0 5px 0;
`

const CompanyStyled = styled.h3`
  color: white;

  text-align: center;
  margin: 5px 0;
`

const DetailListStyled = styled.ul`
  list-style: none;
  padding: 0;
`

const DetailItemContainer = styled.p`
  background: white;
  border-radius: 30px;
  padding: 10px 0 10px 30px;
`

const DetailItemHeading = styled.h4`
  color: white;
  margin: 5px 0 5px 0;
  padding-left: 30px;
  padding-top: 10px;
`

const DetailItemContent = styled.p`
  margin: 10px 0;
`
