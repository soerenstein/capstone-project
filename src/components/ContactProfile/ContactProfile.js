import React from 'react'
import ProfileContent from './ProfileContent'

const contacts = {
  firstName: 'Max',
  lastName: 'Mustermann',
  company: 'Universal GmbH',
  phone: '0176-123123123',
  mail: 'max.mustermann@universal.de',
  street: 'ABC-Stra√üe 105',
  zip: '20239',
  city: 'Hamburg',
  note: 'Wv Montag',
  category: 'pot. Kunde',
}

export default function ContactProfile() {
  return <ProfileContent contacts={contacts} />
}
