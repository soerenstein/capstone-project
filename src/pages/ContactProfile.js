import React from 'react'
import ProfileContent from '../components/ContactProfile/ProfileContent'

export default function ContactProfile({
  contacts,
  onFavoriteClick,
  favorites,
}) {
  return (
    <ProfileContent
      contacts={contacts}
      onFavoriteClick={onFavoriteClick}
      favorites={favorites}
    />
  )
}
