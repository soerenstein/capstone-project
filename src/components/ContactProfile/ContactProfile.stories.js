import React from 'react'
import ContactProfile from './ContactProfile'
import CardWrapper from '../Design/CardWrapper/CardWrapper'

export default {
  title: 'ContactProfile',
  component: ContactProfile,
}

const Profile = (args) => <ContactProfile {...args} />

export const ProfilePage = Profile.bind({})
ProfilePage.args = {}

const Content = (args) => <CardWrapper />

export const ContentItem = Content.bind({})
