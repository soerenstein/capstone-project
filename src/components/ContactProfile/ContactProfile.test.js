import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ContactProfile from './ContactProfile'

describe('The profile', () => {
    it('renders correctly', () => {
        const form = render(<ContactProfile />)
        expect(form).toMatchSnapshot()
    })  
})

describe('The heading', () => {
    it('for contact information renders correctly', () => {
        const {getByText} = render(<ContactProfile  />);  
        expect(getByText('Kontaktdaten')).toBeInTheDocument()
        
     })
     it('for adress renders correctly', () => {
        const {getByText} = render(<ContactProfile  />);  
        expect(getByText('Adresse')).toBeInTheDocument()
        
     })
     it('for category renders correctly', () => {
        const {getByText} = render(<ContactProfile  />);  
        expect(getByText('Kategorie')).toBeInTheDocument()
        
     })
     it('for notes renders correctly', () => {
        const {getByText} = render(<ContactProfile  />);  
        expect(getByText('Notiz')).toBeInTheDocument()
        
     })
})

