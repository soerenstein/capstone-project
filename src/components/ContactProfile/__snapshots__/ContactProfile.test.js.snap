// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The profile renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ProfileContent__ProfileWrapper-sc-19q8044-0 jpCFxJ"
      >
        <h2
          class="ProfileContent__NameStyled-sc-19q8044-1 kfIgmp"
        >
          Max Mustermann
        </h2>
        <h3
          class="ProfileContent__CompanyStyled-sc-19q8044-2 caBkeZ"
        >
          Universal GmbH
        </h3>
        <ul
          class="ProfileContent__DetailListStyled-sc-19q8044-3 coUUXN"
        >
          <h4
            class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
          >
            Kontaktdaten
          </h4>
          <li
            class="ProfileContent__DetailItemContainer-sc-19q8044-4 ddgpiS"
          >
            <p
              class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
            >
              0176-123123123
            </p>
            <p
              class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
            >
              max.mustermann@universal.de
            </p>
          </li>
          <h4
            class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
          >
            Adresse
          </h4>
          <li
            class="ProfileContent__DetailItemContainer-sc-19q8044-4 ddgpiS"
          >
            <p
              class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
            >
              ABC-Straße 105
            </p>
            <p
              class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
            >
              20239
            </p>
            <p
              class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
            >
              Hamburg
            </p>
          </li>
          <h4
            class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
          >
            Kategorie
          </h4>
          <h4
            class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
          >
            Notiz
          </h4>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ProfileContent__ProfileWrapper-sc-19q8044-0 jpCFxJ"
    >
      <h2
        class="ProfileContent__NameStyled-sc-19q8044-1 kfIgmp"
      >
        Max Mustermann
      </h2>
      <h3
        class="ProfileContent__CompanyStyled-sc-19q8044-2 caBkeZ"
      >
        Universal GmbH
      </h3>
      <ul
        class="ProfileContent__DetailListStyled-sc-19q8044-3 coUUXN"
      >
        <h4
          class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
        >
          Kontaktdaten
        </h4>
        <li
          class="ProfileContent__DetailItemContainer-sc-19q8044-4 ddgpiS"
        >
          <p
            class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
          >
            0176-123123123
          </p>
          <p
            class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
          >
            max.mustermann@universal.de
          </p>
        </li>
        <h4
          class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
        >
          Adresse
        </h4>
        <li
          class="ProfileContent__DetailItemContainer-sc-19q8044-4 ddgpiS"
        >
          <p
            class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
          >
            ABC-Straße 105
          </p>
          <p
            class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
          >
            20239
          </p>
          <p
            class="ProfileContent__DetailItemContent-sc-19q8044-6 fVJJvY"
          >
            Hamburg
          </p>
        </li>
        <h4
          class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
        >
          Kategorie
        </h4>
        <h4
          class="ProfileContent__DetailItemHeading-sc-19q8044-5 hLDmSe"
        >
          Notiz
        </h4>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
